internal/store/device.go

```diff
@@ -490,22 +490,17 @@ func (s *DeviceStore) UpdateSummaryStatusBatch(ctx context.Context, orgId uuid.U
 	// https://www.postgresql.org/docs/current/functions-json.html
 	// jsonb_set(target jsonb, path text[], new_value jsonb, create_missing boolean)
 	createMissing := "false"
-	query := fmt.Sprintf(`
-        UPDATE devices
-        SET 
-            status = jsonb_set(
-                jsonb_set(status, '{summary,status}', '"%s"', %s), 
-                '{summary,info}', '"%s"'
-            ),
-            resource_version = resource_version + 1
-        WHERE name IN (%s)`, status, createMissing, statusInfo, tokens)
-
-	args := make([]interface{}, len(deviceNames))
-	for i, name := range deviceNames {
-		args[i] = name
-	}
-
-	return s.db.WithContext(ctx).Exec(query, args...).Error
+	return s.db.WithContext(ctx).
+		Model(&model.Device{}).
+		Where("name IN ?", deviceNames).
+		UpdateColumns(map[string]interface{}{
```

Comment:
I argue that this makes it harder to read. what is the trade-off to making the store more dependent on gorm syntax? I agree that all statements should be evaluated for injection.

Reply:
* It will create a parameterized query, so each parameter will be validated against SQL injection, for example.
* It's more readable as you can see an intention (update columns), and table and where parts, which is not that easy to see in a printf-based raw SQL.

Reply:
Let's hope future code reviews will cover this problem :)